src/api/ledger/transaction.js

```diff
@@ -62,11 +62,19 @@ function getTransactionAsync(identifier: string, options: TransactionOptions,
       error = new errors.NotFoundError('Transaction not found');
     }
 
+    const minLedgerVersion = options.minLedgerVersion;
+
     if (error instanceof errors.NotFoundError
-        && !utils.hasCompleteLedgerRange(remote,
-            options.minLedgerVersion, maxLedgerVersion)) {
-      callback(new errors.MissingLedgerHistoryError('Transaction not found,'
-        + ' but the server\'s ledger history is incomplete'));
+        && !utils.hasCompleteLedgerRange(remote, minLedgerVersion,
+                                         maxLedgerVersion)) {
+      if (minLedgerVersion && maxLedgerVersion
+          && utils.isNonContiguousLedgerRange(remote, minLedgerVersion, 
+                                         maxLedgerVersion)) {
+        callback(new errors.NonContiguousLedgerHistoryError());
+      } else if (maxLedgerVersion 
+                 && utils.isPendingLedgerVersion(remote, maxLedgerVersion)) {
+        callback(new errors.PendingLedgerError());
+      }
```

Comment:
Isn't that this case? https://github.com/ripple/ripple-lib/pull/548/files#diff-c9b59b930eec9b6808a63ab2fbfab346R86

It's just a NotFoundError in that case. If you don't specify a range, or a max